### PR TITLE
handle corrupt images

### PR DESCRIPTION
**Pull request by afiaka87**
Adds error handling for corrupted images/"truncated pngs"

### CODE REVIEW

File: train_dalle.py [hunk 1]
```diff
@@ -204,15 +205,19 @@ def __getitem__(self, ind):
         key = self.keys[ind]
         text_file = self.text_files[key]
         image_file = self.image_files[key]
+        try:
+          image = Image.open(image_file)
+          image_tensor = self.image_tranform(image)
+        except (PIL.UnidentifiedImageError, OSError): # Catch "truncated pngs" and other corrupted images.
+          print(f'{image_file} was corrupt. Skipping.')
+          return self.__getitem__(ind+1)
```
Comment (ex4sperans) [inline train_dalle.py:213]: Won't it break when `ind+1 == len(dataset)`? I think something like `(ind+1) % len(dataset)` should be safer. What do you think?

**Comment by lucidrains** [comment]
this looks pretty good https://github.com/msamogh/nonechucks

**Comment by afiaka87** [comment]
yep that's way better and works with any type of dataset. implementing that instead @lucidrains - good find.

**Comment by afiaka87** [comment]
this is out of scope for this issue i.e. i'm being lazy, but the "Transforms as Filters" thing would be perfect for multilingual datasets. hopefully we can get a cool little pipeline out of it later on.

**Pull request closed without merging**